prediction_market_agent/agents/crewai_subsequential_agent/deploy.py

```diff
@@ -0,0 +1,80 @@
+import os
+import random
+from decimal import Decimal
+
+from langchain_openai import OpenAI
+from prediction_market_agent_tooling.deploy.agent import DeployableAgent
+from prediction_market_agent_tooling.markets.agent_market import AgentMarket
+from prediction_market_agent_tooling.markets.data_models import BetAmount, Currency
+from prediction_market_agent_tooling.markets.markets import MarketType
+
+from prediction_market_agent.agents.crewai_subsequential_agent.crewai_agent_subquestions import (
+    CrewAIAgentSubquestions,
+)
+from prediction_market_agent.agents.known_outcome_agent.known_outcome_agent import (
+    Result,
+)
+from prediction_market_agent.agents.utils import market_is_saturated
+
+
+class DeployableThinkThoroughlyAgent(DeployableAgent):
+    # For cheaper credits at this experimental stage
+    model = "gpt-4-turbo-preview"
+
+    def load(self) -> None:
+        self.markets_with_known_outcomes: dict[str, Result] = {}
+
+    def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
+        # We simply pick 5 random markets to bet on
+        picked_markets: list[AgentMarket] = []
+        random.shuffle(markets)
+        for market in markets:
+            # Assume very high probability markets are already known, and have
+            # been correctly bet on, and therefore the value of betting on them
+            # is low.
+            if not market_is_saturated(market=market):
+                picked_markets.append(market)
+                if len(picked_markets) == 5:
+                    break
+
+        return picked_markets
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        # The answer has already been determined in `pick_markets` so we just
+        # return it here.
+        os.environ["OPENAI_MODEL_NAME"] = "gpt-4-turbo-preview"
```

Comment:
What's this for?

Reply:
For setting the OpenAI model using crewai, the way it's done is via ENV -> https://docs.crewai.com/how-to/LLM-Connections/#crewai-agent-overview

Reply:
What do you think about doing it like they do here, please?https://github.com/joaomdmoura/crewAI/blob/66b77ffd080a8acf48917bd2e329969b5d9c486a/tests/task_test.py#L220 :

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/065bebe9-0d62-4221-a2c9-19ec036185e1)

Inside of `CrewAIAgentSubquestions`'s `__init__` method and the currently unused `openai_model_name` variable?


The problem I see is that it will be error-prone to use this class then. Because if it changes such environment variable, then all the other agents are changed to it as well. For example, in benchmarks:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/69be9071-dad6-441c-bec5-0a4cab3bde53)

Reply:
I see you passed in `llm` arg, is `function_calling_llm` unused?